### DISCUSSION TITLE
Should we refer to the recipient as the owner?

### DISCUSSION THREAD

**Discussion by PaulRBerg** [sablier-labs/evm-monorepo#141] [General]
In #131, @andreivladbrg made all of our contract inherit from the ERC-721 standard. The most crucial implication of that is that now there is no `recipient` field in the `Stream` struct anymore - the [`ownerOf`](https://eips.ethereum.org/EIPS/eip-721#specification) function of the ERC-721 is what gives the recipient of the stream now.

We have a terminology problem. We can keep referring to the recipient of the stream as the "recipient" in the code, too, by choosing to keep the `getRecipient` getters and by naming our in-memory variables as such.

However, that incurs a little bit of additional gas (though I am not sure if it's only deployment cost that get affected or there's also a runtime overhead).

It would certainly be more gas efficient to use the `ownerOf` function directly and remove all `recipient` references. But that would create some confusion because the sender of the stream could easily be confused with the ERC-721 owner.

I'm leaning towards keeping the `recipient`, since the gas overhead is small while the clarity benefits are large. But I opened this discussion to document my thinking and let you all share your feedback if you disagree.

**Comment by andreivladbrg** [comment]
No disagreement here, we should keep the `getRecipient` function.

**Reply by PaulRBerg** [reply]
Cool.
